### ISSUE TITLE
Incorrect scaling in native Gnome 47 UI with fractional scaling enabled

### ISSUE THREAD

**Issue by shawnyeager** [catppuccin/cursors#45]
I installed via Arch AUR, running Gnome 47. In apps like Firefox, the cursors appears to render correctly, however in Gnome UI, Files/Nautilus, etc., the cursors appear to be 2x size and heavily pixelated.

**Comment by sgoudham** [comment]
Hey 👋,

That's disappointing to hear since v1.0.0 was meant to fix all the issues on GNOME. Could you provide a screen recording highlighting what happens when the cursor moves from Firefox to something like Files/Nautilus? 

We're making use of KDE Breeze's scripts for generating the scales so it may be a problem in that script we need to fix or something else entirely.

**Comment by ARKye03** [comment]
> I installed via Arch AUR, running Gnome 47. In apps like Firefox, the cursors appears to render correctly, however in Gnome UI, Files/Nautilus, etc., the cursors appear to be 2x size and heavily pixelated.

+1 I support this, on Hyprland the `catppuccin-cursors-mocha` pkg from AUR seems off completely, with a size of 24 it's too big, and 20 it's too small, however, I can work with 24, yet with a size of 20 it looks pixelated and incomplete(some borders of the cursor won't render they appear `cut`. I set the cursor via `hyprctl setcursor "catppuccin-mocha-dark-cursors" 20`

**Comment by sgoudham** [comment]
If you're on Hyprland, scaling is a known issue (#31) because I believe the current implementation uses PNGs in order to work around a different bug raised a while back. @kerichdev can provide more context about that.

**Comment by ARKye03** [comment]
> If you're on Hyprland, scaling is a known issue (#31) because I believe the current implementation uses PNGs in order to work around a different bug raised a while back. @kerichdev can provide more context about that.

Oh, I see, yes that seems to be related to my comment above...

**Comment by shawnyeager** [comment]
Thanks, @sgoudham. I've tried to record using Gnome's built in recorder, as well as flameshot, but neither one capture the pointer, oddly enough (see below). Any ideas on how to capture?

[Screencast From 2024-10-25 09-21-37.webm](https://github.com/user-attachments/assets/755848c6-1650-456c-839c-8367a3cb3a81)

**Comment by shawnyeager** [comment]
@Covkie No other cursor set exhibits this behavior, or I might agree.

**Comment by jinliu** [comment]
This is a known GTK 4.16 bug. You can find explanation in the end of this article:

https://blogs.kde.org/2024/10/09/cursor-size-problems-in-wayland-explained/

It's triggered when you have scaling, and a cursor theme whose nominal size != image size.

It's fixed in GTK but not backported to 4.16.

If Catppuccin use a nominal size 32 instead of 24, it would work around the problem.

**Comment by shawnyeager** [comment]
I'm running 4.16.3 on Arch, so that didn't fix it for me.

**Comment by shadow-absorber** [comment]
does 200% scaling count as fractional scaling as have the same problem here using 200% scaling on hyprland on a 4k monitor on arch linux

**Comment by Covkie** [comment]
GNOME 48 along with GTK 4.18 has reached stable.
gtk 4.18 has the client side rendering issue that jinliu mentioned fixed. It also has support for the wayland `cursor-shape` protocol which will fix cursor inconsistencies in GTK apps on other DEs


Update your gtk 4 package and this should be resolved.

**Comment by shawnyeager** [comment]
@Covkie Tried this yesterday. Unfortunately, it is not.

**Comment by Covkie** [comment]
Seems entirely to be a bug in mutter
https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4345

Edit 1: The arch mutter maintainer applied that patch in 48.0-2 https://gitlab.archlinux.org/archlinux/packaging/packages/mutter/-/commit/5b1de49a69c3ce93dcc445d8fd4dbe84a83b305c

If you tested it yesterday I'm unsure why you still experienced the issue.

Edit 2: I'm on gnome wayland 48.0-3 and gtk 4.18.2-1 and I cannot repro this like how i was before. I'm unsure what you are experiencing.

Can you show the gsettings results for cursor-theme & cursor-size as well as screenshots or a screen capture of the size issues?

**Comment by shawnyeager** [comment]
Thanks. I'm seeing a host of odd renderings problems since upgrading to Gnome 48, most notable anything Electron-related.

**Comment by sgoudham** [comment]
Are you happy for me to close this issue as it seems the problem is no longer directly related to Catppuccin? @shawnyeager

**Comment by shawnyeager** [comment]
Thanks, @sgoudham. I'm unsure (lack the technical depth) why others cursors like Adwaita don't suffer this same problem, so I'll defer to you.

**Comment by sgoudham** [comment]
> Thanks, @sgoudham. I'm unsure (lack the technical depth) why others cursors like Adwaita don't suffer this same problem, so I'll defer to you.

I'm not particularly technical regarding cursors either. I was going off @Covkie's earlier comment as they are seemingly unable to reproduce your issue and therefore was hoping that it is no longer our cursors that are the problem.

**Comment by Covkie** [comment]
@sgoudham I'm pretty sure this issue is resolved (gtk stable release has it patched) but I can't say for certain unless @shawnyeager can show logs and images of what they have set and what the visual issues are..

**Comment by shawnyeager** [comment]
@Covkie The problem appears when scaling (fractional or otherwise) is active. 

On my Framework 13", I have 175% scaling on for the native display. I've also tested 200%. On this machine, cursors are magnified and pixelated in line with scale magnification. 

On my trusty, old Thinkpad X1 Carbon Gen 8, with no scaling, cursors always render correctly. 

Both are running Arch, fully updated, including Gnome 48.

**Comment by shadow-absorber** [comment]
fixed for shadow too on
```
local/gtk4 1:4.18.2-1
    GObject-based multi-platform GUI toolkit
```
using hyprland with 200% scaling.. relevant options in hyprland.conf is:
```
exec-once = hyprctl setcursor catppuccin-mocha-dark-cursors 32
exec-once = gsettings set org.gnome.desktop.interface cursor-theme catppuccin-mocha-dark-cursors
exec-once = gsettings set org.gnome.desktop.interface cursor-size 32

exec-once = xrdb $HOME/.Xresources

# Some default env vars.
env = XCURSOR_SIZE,32
env = XCURSOR_THEME,catppuccin-mocha-dark-cursors
```

**Comment by Covkie** [comment]
@shawnyeager Just to confirm your issue is that the cursors are blurry but they stay the same size across applications?

What cursor size & screen scale do you have set? can you please show the gsettings result for cursor-size and show screenshots of the blurriness

I believe this issue is fixed and shawnyeager is experiencing a different problem that cannot be fixed sanely on our end.

@sgoudham

**Comment by shawnyeager** [comment]
@Covkie No, they change sizes when moving in and out of window scope for certain applications. Unfortunately, this is where my depth fails me. I can say, for example, that the problem is visible with Electron-based apps.

I'd grab screenshots, but I've already removed the package. Sorry.

As mentioned, I can confirm that neither Yaru nor Adwaita cursors exhibit this behavior.
